service/geopmdpy/system_files.py

```diff
@@ -679,10 +682,42 @@ def remove_batch_server(self, client_pid):
             RuntimeError: Client does not have an open session
 
         """
+        batch_pid = self._sessions[client_pid]['batch_server']
+
         self.check_client_active(client_pid, 'remove_batch_server')
         self._sessions[client_pid].pop('batch_server')
         self._update_session_file(client_pid)
-        # TODO: Unlink any shared memory files left around
+
+        # Unlink any temporary files left around from the Batch Server
+        # This includes shared memory files, fifo files, and the lock file.
+        if (os.path.exists(self._LOCK_PATH)):
```

Comment:
Cleaning the lock file is done when the session ends (line 687).  That code checks if the session that obtained the lock file is still alive, and it is not it is removed.  The code here will remove the lock file regardless of which session owns it.  This may remove a lock created by a process other than the batch server.  The removal of this file must depend on the PID that has ended.